### PR TITLE
Launch gitversion error message

### PR DESCRIPTION
**Pull request by gtarpenning**
Partially Addresses WB-9990

Description
-----------
Updates error messages to provide more explicit handling when cloning from a git repo fails. This is usually due to an incorrect branch, either through a typo in the `git-version` parameter or because the there is no `master` branch.

As GitHub has changed their default branch to `main`, this error will likely increasingly occur, thus better handling and error messages seem appropriate. Eventually we might want to change the default branch from `master` -> `main`.

Testing
-------
Multiple variations of branch configs on a remote git repo.

### CODE REVIEW

File: wandb/sdk/launch/utils.py [hunk 1]
```diff
@@ -430,12 +430,26 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
...
             raise ExecutionError(
                 "Unable to checkout version '%s' of git repo %s"
-                "- please ensure that the version exists in the repo. "
+                "- please ensure that the version exists in the repo. \n"
                 "Error: %s" % (version, uri, e)
             )
     else:
-        repo.create_head("master", origin.refs.master)
-        repo.heads.master.checkout()
+        version = "master"  # change to "main" for current github practice?
```
Comment (KyleGoyette) [inline wandb/sdk/launch/utils.py:437]: Lets add a master/main fallback. Start with main and fallback to master. That way we prefer the current practice but dont drop support for older repos

File: wandb/sdk/launch/utils.py [hunk 2]
```diff
@@ -424,20 +424,28 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
...
             raise ExecutionError(
                 "Unable to checkout version '%s' of git repo %s"
-                "- please ensure that the version exists in the repo. \n"
+                "- please ensure that the version exists in the repo. "
                 "Error: %s" % (version, uri, e)
             )
     else:
-        version = "master"  # change to "main" for current github practice?
+        # Check if main is in origin, else set branch to master
+        branches = [ref.name for ref in repo.references]
```
Comment (gtarpenning) [inline wandb/sdk/launch/utils.py:439]: if repo doesn't exist repo.references might error here. Is it possible to reach this point with a broken/nonexistent git repo?

File: wandb/sdk/launch/utils.py [hunk 3]
```diff
@@ -424,7 +424,8 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
     repo = git.Repo.init(dst_dir)
     origin = repo.create_remote("origin", uri)
     origin.fetch()
-    if version is not None:
+
+    if version:
```
Comment (hu-po) [inline wandb/sdk/launch/utils.py:428]: nit: `is not None` is a more explicit check imo: `bool("") == ("" is not None)`

File: wandb/sdk/launch/utils.py [hunk 4]
```diff
@@ -434,8 +435,30 @@ def _fetch_git_repo(dst_dir: str, uri: str, version: Optional[str]) -> None:
...
+            version = "main"
+        else:
+            version = "master"
+
+        try:
+            repo.create_head(version, origin.refs[version])
+            repo.heads[version].checkout()
+            wandb.termlog(f"No git branch passed. Defaulted to branch: {version}")
+        except git.exc.GitCommandError as e:
+            raise ExecutionError(
```
Comment (hu-po) [inline wandb/sdk/launch/utils.py:450]: Do we want `LaunchErrors` here instead of `ExecutionErrors`? @KyleGoyette

Comment (KyleGoyette) [reply wandb/sdk/launch/utils.py:450]: Yep, throw a launch error here, the agent catches it better.

**Review by KyleGoyette — COMMENTED** [review]
LGTM, but lets write a test using both master and main so we'll know if this code path breaks.

**Review by KyleGoyette — APPROVED** [review]
🚢

**Pull request merged**